compiler-rt/lib/asan/asan_allocator.cpp

```diff
@@ -1117,6 +1111,99 @@ uptr asan_malloc_usable_size(const void *ptr, uptr pc, uptr bp) {
   return usable_size;
 }
 
+namespace {
+
+void *asan_new(uptr size, BufferedStackTrace *stack, bool array) {
```

Comment:
naming is hard, but `array` doesn't immediately register to me as a boolean :-) . Should we consider something like `isArray` to make it more obvious?

Reply:
Considering its isolated to just this file and in an anonymous namespace, it's less important to be that clear. If this were in a header and accessible across many files, I would be inclined to agree.

Reply:
I'd prefer we move it into name -> asan_new_array()

Reply:
@vitalybuka do you mean drop the array arg and split into two functions asan_new_array and asan_new_scalar or something like that?